crates/wdl-analysis/src/types.rs

```diff
@@ -911,6 +949,134 @@ impl Coercible for StructType {
     }
 }
 
+/// Represents the type of an enum.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct EnumType {
+    /// The name of the enum.
+    name: Arc<String>,
+    /// The common coerced type computed from all variant values.
+    value_type: Arc<Type>,
```

Comment:
This can just be `Type`.

A reminder to myself: I need to clean these type definitions up a little.